lib/screens/signup_screen.dart

```diff
@@ -4,13 +4,44 @@ import 'package:font_awesome_flutter/font_awesome_flutter.dart';
 import 'package:scdao_mobile/constants/color_constants.dart';
 import 'package:email_validator/email_validator.dart';
 import 'dart:developer';
+import 'package:http/http.dart' as http;
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'dart:convert';
+import 'dart:io';
 
-class SignupScreen extends StatefulWidget {
+class SignupScreen extends ConsumerStatefulWidget {
   @override
   _SignupScreenState createState() => _SignupScreenState();
 }
 
-class _SignupScreenState extends State<SignupScreen> {
+final signupRepositoryProvider =
+    Provider<SignupRepository>((ref) => SignupRepository());
+
+class SignupRepository {
+  var url = Uri.parse("http://192.168.0.44:8888/api/signup");
+  Future<http.Response> signup(String username, String password) async {
+    Map<String, String> bodyParams = new Map();
+    bodyParams["username"] = username;
+    bodyParams["password"] = password;
+    Map<String, String> headersMap = new Map();
+    headersMap["content-type"] = "application/x-www-form-urlencoded";
+    final http.Response res = await http
+        .post(url, headers: headersMap, body: bodyParams, encoding: Utf8Codec())
+        .then((res) {
+      if (res.statusCode == 200) {
+        print(res.body);
+        return res;
+      } else {
+        print("error");
+        return res;
+      }
+    });
+
+    return res;
+  }
+}
```

Comment:
You might also consider using `FutureProvider` as it handles all the states more easily. 

https://riverpod.dev/docs/providers/future_provider/